src/lando/pushlog/models/push.py

```diff
@@ -0,0 +1,73 @@
+from django.db import models
+
+from lando.main.models import Repo
+
+from .commit import Commit
+from .consts import MAX_BRANCH_LENGTH, MAX_URL_LENGTH
+
+PUSH_SCM_TYPE_GIT = "git"
+PUSH_SCM_TYPES = [PUSH_SCM_TYPE_GIT]
+
+
+class Push(models.Model):
+    push_id = models.PositiveIntegerField()
+
+    repo = models.ForeignKey(
+        Repo,
+        # We don't want to delete the PushLog, even if we were to delete the repo
+        # object.
+        on_delete=models.DO_NOTHING,
+    )
+
+    # Full URL to the upstream repository, to allow re-linking the data if anything goes
+    # wrong in the DB.
+    repo_url = models.CharField(max_length=MAX_URL_LENGTH, db_index=True)
+
+    # Branch names are limited by how long a filename the filesystem support.
+    branch = models.CharField(max_length=MAX_BRANCH_LENGTH, db_index=True)
+
+    date = models.DateField(
+        auto_now=False,
+        auto_now_add=True,
+        db_index=True,
+    )
+
+    # Maximum total lengths are defined in RFC-5321 [0]: 64 for the local-part, and 255
+    # for the domain.
+    # [0] https://datatracker.ietf.org/doc/html/rfc5321#section-4.5.3.1.1
+    user = models.EmailField(max_length=64 + 1 + 255)
+
+    commits = models.ManyToManyField(Commit)
```

Comment:
Would we want to keep actual SHA in this object? Same as we keep the repo_url?

Reply:
A commit can be pushed to multiple branches independently, so I think we may find a commit in more than one push.

What I meant with the SHA question was whether we should use the SHA as a database key, rather than an additional internal column. The pro is that the push table on its own would contain useful information to reconcile commits, even with incomplete information. The con are that the column will be 160bit, and that it would diverge from standard DB models. Thinking more about it, it's probably getting too corner case to be worth it.